### PR TITLE
Add new component skeleton table

### PR DESCRIPTION
**Pull request by karelhala**

### CODE REVIEW

**Comment by ryelo** [comment]
@karelhala we recently had discussions about loading states. From what we talked about, the skeleton is only supposed to be used if you know exactly what data is going to be there. In this scenario, we don't know the rows or columns, so there should only be a spinner inside an empty table

File: src/PresentationalComponents/Pagination/PaginationNav.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ const PaginationNav = ({
                 min="1"
                 data-action="set-page"
                 max={ amountOfPages }
-                style={ { width: `${lastPage.toString().length}rem` } }
+                style={ { width: `${lastPage.toString().length}rem`, minWidth: '3rem' } }
```
Comment (AllenBW) [inline src/PresentationalComponents/Pagination/PaginationNav.js]: yes please 🤗

**Comment by karelhala** [comment]
> From what we talked about, the skeleton is only supposed to be used if you know exactly what data is going to be there. In this scenario, we don't know the rows or columns, so there should only be a spinner inside an empty table

Well this skeleton table is for tables you know how many columns they have. So you kinda know how big each column will be. And you also know how many items you are fetching

**Comment by ryelo** [comment]
@karelhala Yeah, that makes sense. I think this was more for the rows. If we show 10 rows and only one shows up, that will be breaking the (future) rules. We would use a spinner in that case.

**Comment by karelhala** [comment]
@ryelo oh, okay that makes sense. What would be the optimal path here? Wait until we figure out spinner? Or merge this and update once we get the spinner and such out?

**Comment by ryelo** [comment]
@karelhala Spinner is supposed to be implemented in react by March 22. I can approve this for now, or we can wait until the spinner is implemented. Doesn't matter to me.

**Comment by karelhala** [comment]
@ryelo well we can merge this one and deprecate/update later on when PF implements spinner.

**Comment by ryelo** [comment]
@karelhala WFM!

File: src/PresentationalComponents/SkeletonTable/SkeletonTable.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        return (
+            <Table cells={ this.createColumns() } rows={ this.createRows() } aria-label="Loading">
+                <TableHeader />
+                <TableBody />
+            </Table>
+        );
+    }
+}
+
+SkeletonTable.propTypes = {
```
Comment (ryelo) [inline src/PresentationalComponents/SkeletonTable/SkeletonTable.js]: Can we give the user an option to pass in their headers?

File: src/PresentationalComponents/SkeletonTable/SkeletonTable.test.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import SkeletonTable from './SkeletonTable';
+
+describe('SkeletonTable component', () => {
+    it('should render correctly', () => {
+        const wrapper = shallow(<SkeletonTable colSize={5} rowSize={15}/>)
+        expect(toJson(wrapper)).toMatchSnapshot();
+    })
+
+    it('should render correctly without rows', () => {
+        const wrapper = shallow(<SkeletonTable colSize={5} rowSize={15} />)
```
Comment (ryelo) [inline src/PresentationalComponents/SkeletonTable/SkeletonTable.test.js]: Do you want to remove the `rowSize` prop for this one?

**Pull request merged**